app/services/edge_cache/service.rb

```diff
@@ -0,0 +1,115 @@
+module EdgeCache
+  class Service
+    PROVIDERS = %w[fastly nginx].freeze
+
+    def initialize(path)
+      # TODO: (Vaidehi Joshi) - Right now, we are checking that nginx is
+      # available on every purge request/call to this service. If we are going
+      # to bust multiple paths, we should be able to check that nginx is
+      # available just once, and persist it on the class with @provider_available?.
+      # Then, we could allow for an array of @paths = [] to be passed in,
+      # and on single service instance could bust multiple paths in order.
```

Comment:
I'll probably tackle this in a separate PR. Just want to make sure this works as expected on production before making _too many_ changes.